### PR TITLE
Use `clang-tidy -fix` to replace 0 by nullptr

### PR DESCRIPTION
**Pull request by lrineau**
## Summary of Changes

As a followup to #3979, and because I hear a lot about clang-tidy, I wanted to try and use clang-tidy, to see if one can automatically replace `0` or `NULL` by `nullptr`. I have created a tiny `.clang-tidy` file.

The only check applied for now is [`modernize-use-nullptr`](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-nullptr.html).

To use it:
```shell
cmake -DCMAKE_CXX_CLANG_TIDY="clang-tidy;-fix" build
```
and then I have used:
```shell
make && ctest -L Mesh_2
```
to compile (with clang-tidy enabled) the whole testsuite of Mesh_2. The option `-fix` of clang-tidy tells it to modify the source code directly. And then I have checked the results because committing: all was good. But I have modified the strange modifications made by #3979 in `Hash_map/include/CGAL/Tools/chained_map.h` (`NULLKEY` had been modified to `nullptrKEY` by error).

## Release Management

* Affected package(s): Mesh_2 and dependencies
* License and copyright ownership: maintenance by GeometryFactory

I am not 100% sure that we want to apply that sort of pull-request to `master` for CGAL-5.0, because, like PRs #3979, #3978, and #3975, that increases the chance of conflicts with bug-fix pull-requests based on CGAL-4.13 or CGAL-4.14.

### CODE REVIEW

**Comment by MaelRL** [comment]
What is missing to this PR?

**Comment by lrineau** [comment]
There is a lot that is missing in this PR. I have applied clang-tidy only to Mesh_2 and its dependencies. I created that issue to start a discussion: do we want that tool clang-tidy applied to CGAL as a whole? And if yes, which rules of clang-tidy? So far I have experimented only with `modernize-use-nullptr`.

**Comment by MaelRL** [comment]
A possible path could be to extend this PR to all packages but just for `nullptr` and not include any `.clang_tidy` / `NOLINTNEXTLINE` so that it can be integrated. Afterwards, we could open an issue and look at all clang tidy options.

**Comment by lrineau** [comment]
I agree. We should coordinate with other massive PR, like #3895.

**Comment by maxGimeno** [comment]
@lrineau is this PR still needed ? If yes, then it needs an update.

**Comment by lrineau** [comment]
> @lrineau is this PR still needed ? If yes, then it needs an update.

It is not "needed". That is just an improvement on CGAL code.

But it is useful.



@MaelRL https://github.com/CGAL/cgal/pull/4041#issuecomment-581814170:
> A possible path could be to extend this PR to all packages but just for `nullptr` and not include any `.clang_tidy` / `NOLINTNEXTLINE` so that it can be integrated. Afterwards, we could open an issue and look at all clang tidy options.

Why do you want to remove the `/.clang-tidy` file? clang-tidy could be a new useful tool in our testsuite.

**Comment by maxGimeno** [comment]
ping @MaelRL

**Comment by MaelRL** [comment]
I just meant: it could be ignored temporarily so that there is no extra pollution in the results for warnings that we know we can first fix "off-testsuite". But sure, it would be useful to have.

File: STL_Extension/include/CGAL/Handle.h [hunk 1]
```diff
@@ -42,14 +42,16 @@ class Handle
   public:
 
     typedef std::ptrdiff_t Id_type ;
+    Handle()
+    : PTR(static_cast<Rep*>(nullptr)) {}
 
     Handle() noexcept
         : PTR(static_cast<Rep*>(0)) {}
```
Comment (lrineau) [inline STL_Extension/include/CGAL/Handle.h:49]: Bad merge: now you have twice the default-constructor.

**Comment by maxGimeno** [comment]
Conflicts with master

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-106.shtml

**Pull request merged**